pkg/repo/chartrepo.go

```diff
@@ -156,46 +118,6 @@ func (r *ChartRepository) DownloadIndexFile() (string, error) {
 	return fname, os.WriteFile(fname, index, 0644)
 }
 
-// Index generates an index for the chart repository and writes an index.yaml file.
-func (r *ChartRepository) Index() error {
```

Comment:
My only concern is removing this non-deprecated public function, even if we don't use it internally. Do we have an alternative recommendation for people who may be using it out in the wild?

Reply:
`(r *ChartRepository) Load() error {` is deprecated, and is the only thing that writes to `ChartPaths()`. So in effect, removing that (deprecated) function, but not this would be quite awkward for users.

Reply:
`repo.IndexDirectory()` will take a directory and generate the index. This is what the `helm repo index` command uses. The code path being used here isn't used because of the alternative path.

The code path this PR is removing is problematic. For example, right after this `r.generateIndex()` is called. That function uses `loader.Load` which can load a chart as a directory. This is something that should not happen.

I can understand having an internal representation that can generate an index file from a known set of charts. I would want OCI charts included in that. This cannot handle those situations.

I did some searching, using GitHub search, and I did not find anyone using this section of code.

Reply:
i think this makes sense. we are able to make breaking changes in v4 either way, i just wanted to think it through together. thanks!